### PR TITLE
Click refactor

### PR DESCRIPTION
**Pull request by AndiAJ**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt [hunk 1]
```diff
@@ -96,15 +102,51 @@ object MatcherHelper {
...
     }
 
+    fun assertComposeItemExists(
+        vararg composeItems: SemanticsNodeInteraction,
+        exists: Boolean = true,
+    ) {
+        for (composeItem in composeItems) {
+            if (exists) {
+                Log.i(TAG, "assertComposeItemExists: Trying to verify compose item ${composeItem.printToString()} exists")
+                composeItem.assertExists()
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt:123]: I think this only verifies and syncs to make sure the elements exists in the tree, not if it's actually displayed on the screen. assertIsDisplayed() should do that. Maybe we should have both.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt [hunk 2]
```diff
@@ -96,15 +102,51 @@ object MatcherHelper {
...
+    }
+
+    fun assertEspressoItemIsDisplayed(
+        vararg espressoItems: ViewInteraction,
+        exists: Boolean = true,
+    ) {
+        for (espressoItem in espressoItems) {
+            if (exists) {
+                Log.i(TAG, "assertEspressoItemExists: Trying to verify espresso item $espressoItem is displayed")
+                espressoItem.check(matches(isDisplayed()))
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt:140]: There are many situations where we use a UIObject wait method before doing an espresso assertion or action. One example below: 
image

@jjSDET How should we handle these situations, where we need to make sure we wait long enough for the item to be visible? Should we just replace everything Espresso with UIAutomator? (I don't like that idea, as it turned out to work better sometimes than UI Automator)

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt [hunk 3]
```diff
@@ -142,4 +148,35 @@ object TestHelper {
         assertFalse("Light theme not selected", expected)
 
     fun verifyDarkThemeApplied(expected: Boolean) = assertTrue("Dark theme not selected", expected)
+
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt:151]: I'm not sure this is the right place for these methods. Maybe we should have a MatcherInteractionsHelper class?

**Comment by gabrielluong** [comment]
We have moved our firefox-android repo over to [mozilla-central](https://hg.mozilla.org/mozilla-central). Please feel free to reopen a new PR against the new repo. You will have to clone the new repository and reapply your changes. Thank you again, and sorry for the inconvenience.

For more information, please see https://github.com/mozilla-mobile/firefox-android/wiki#upcoming-migration-to-mozilla-central.

**Pull request closed without merging**